File: Hello_World_Tutorials/testing/c++/circ_dllist.cpp
#include <iostream>
#include <string>

#include "Bridges.h"
#include "CircDLelement.h"

using namespace std;
using namespace bridges;


int main() {
	Bridges::initialize(6, "kalpathi60", "486749122386");
//	Bridges::initialize(6, "bridges_public", "997924677918");

	Bridges::setTitle("An Example Circular Doubly Linked List");
						// create list nodes
	CircDLelement<int> *el0 = new CircDLelement<int>(0, "0");
	CircDLelement<int> *el1 = new CircDLelement<int>(1, "1");
	CircDLelement<int> *el2 = new CircDLelement<int>(2, "2");
	CircDLelement<int> *el3 = new CircDLelement<int>(3, "3");
       					// link nodes 
	el0->setNext(el1);
	el1->setNext(el2);
	el2->setNext(el3);
	el3->setNext(el0);

	el0->setPrev(el3);
	el1->setPrev(el0);
	el2->setPrev(el1);
	el3->setPrev(el2);

						// set color of nodes
/*
	el0->getVisualizer()->setColor("red");
	el1->getVisualizer()->setColor("green");
	el2->getVisualizer()->setColor("blue");
	el3->getVisualizer()->setColor("yellow");
						// set link colors
	el0->getLinkVisualizer(el1)->setColor("cyan");
	el1->getLinkVisualizer(el2)->setColor("magenta");
	el2->getLinkVisualizer(el3)->setColor("green");
						// set link thickness
	el0->getLinkVisualizer(el1)->setThickness(3);
*/
						// provide Bridges handle to data structure
	Bridges::setDataStructure(el0);
						// visualize the data structure
	Bridges::visualize();

	return 0;
}

File: Hello_World_Tutorials/testing/python/circ_dllist.py
from Bridges import *
from StudentInfo import *
from CircDLelement import *



def insertFront(tailElement, newElement):
    tailNextElement = tailElement.get_next()

    newElement.set_next(tailNextElement)
    newElement.set_prev(tailElement)

    tailNextElement.set_prev(newElement)
    tailElement.set_next(newElement)

    return tailElement





class CircDLtutorial:
    bridges = Bridges(6, "YOUR_USER_ID", "YOUR_API_KEY")

    students = []

    students.append(Student("00000000000",
                            "Gretel Chaney",
                            "CS",
                            "g.chaney@generated.com",
                            "magenta",
                            "blue",
                            9.0))

    students.append(Student("00000000001",
                            "Karol Soderman",
                            "SIS",
                            "k.soderman@generated.com",
                            "magenta",
                            "red",
                            11.0))

    students.append(Student("00000000002",
                            "Lamont Kyler",
                            "BIO",
                            "l.kyler@generated.com",
                            "yellow",
                            "green",
                            50.0))

    students.append(Student("00000000003",
                            "Gladys Serino",
                            "CS",
                            "g.serino@generated.com",
                            "green",
                            "magenta",
                            9.0))

    students.append(Student("00000000004",
                            "Starr Mcginn",
                            "CS",
                            "s.mcginn@generated.com",
                            "red",
                            "cyan",
                            15.0))

    head = None

    for i in range(len(students)):
        if i > 0:
            head = insertFront(head, CircDLelement(label = "", e = students[i]))
        else:
            head = CircDLelement(label = "", e = students[i])

    current = head

    for i in range(len(students)):
        current.set_label(current.get_value().getName())
        current.get_visualizer().set_color(current.get_value().getLikeColor())

        current.get_link_visualizer(current.get_next()).set_color(current.get_value().getDislikeColor())
        current.get_link_visualizer(current.get_next()).set_thickness(current.get_value().getCreditHours()*.2)

        current.get_link_visualizer(current.get_prev()).set_color(current.get_value().getDislikeColor())
        current.get_link_visualizer(current.get_prev()).set_thickness(current.get_value().getCreditHours()*.2)

        current = current.get_next()

    bridges.set_data_structure(head)
    bridges.visualize()
